src/cdk-experimental/table/table-virtual-scroll.ts

```diff
@@ -0,0 +1,260 @@
+  }
+
+  stickyEndColumnsUpdated(update: StickyUpdate): void {
+    this._listener?.stickyEndColumnsUpdated(update);
+  }
+
+  stickyFooterRowsUpdated(update: StickyUpdate): void {
+    this._listener?.stickyFooterRowsUpdated(update);
+  }
+
+  stickyHeaderRowsUpdated(update: StickyUpdate): void {
+    this._listener?.stickyHeaderRowsUpdated(update);
+  }
+}
+
+
+/**
+ * A directive that enables virtual scroll for a {@link CdkTable}.
+ */
+@Directive({
+  selector: 'cdk-table[virtualScroll], table[cdk-table][virtualScroll]',
+  exportAs: 'cdkVirtualScroll',
+  providers: [
+    {provide: _VIEW_REPEATER_STRATEGY, useClass: _RecycleViewRepeaterStrategy},
+    // The directive cannot provide itself as the sticky positions listener because it introduces
+    // a circular dependency. Use an intermediate listener as a proxy.
+    {provide: STICKY_POSITIONING_LISTENER, useClass: PositioningListenerProxy},
+    // Initially emit an empty range. The virtual scroll viewport will update the range after it is
+    // initialized.
+    {
+      provide: _TABLE_COLLECTION_VIEWER,
+      useFactory: () => new BehaviorSubject<ListRange>({start: 0, end: 0}),
+    },
+  ],
+  host: {
+    'class': 'cdk-table-virtual-scroll',
+  },
+})
+export class CdkTableVirtualScroll<T> implements CdkVirtualScrollRepeater<T>, OnDestroy, StickyPositioningListener {
+  /** Emits when the component is destroyed. **/
+  private _destroyed = new ReplaySubject<void>(1);
+
+  /** The currently rendered range of indices. */
+  private _renderedRange: ListRange;
+
+  /** Emits when the virtual scroll viewport's CSS `translateY()` value changes. */
+  private _renderedContentOffsetChanged = this._viewport.renderedContentOffsetChanged.pipe(
+      map(() => this._viewport.getOffsetToRenderedContentStart()),
+      filter(offset => offset !== null) as OperatorFunction<number|null, number>,
+      distinctUntilChanged());
+
+  /** Emits when the header rows sticky state changes. */
+  private readonly _headerRowStickyUpdates = new Subject<StickyUpdate>();
+
+  /** Emits when the footer rows sticky state changes. */
+  private readonly _footerRowStickyUpdates = new Subject<StickyUpdate>();
+
+  /**
+   * Observable that emits the data source's complete data set. This exists to implement
+   * {@link CdkVirtualScrollRepeater}.
+   */
+  get dataStream(): Observable<T[] | ReadonlyArray<T>> {
+    return this._table._dataSourceChanges.pipe(
+        switchMap(dataSource => this._getDataStream(dataSource)),
+        shareReplay(1));
+  }
+
+  /**
+   * The size of the cache used to store unused views. Setting the cache size to `0` will disable
+   * caching.
+   */
+  @Input()
+  get viewCacheSize(): number {
+    return this._viewRepeater.viewCacheSize;
+  }
+  set viewCacheSize(size: number) {
+    this._viewRepeater.viewCacheSize = size;
+  }
+
+  constructor(
+      private readonly _table: CdkTable<T>,
+      private readonly _tableEl: ElementRef<CdkTable<T>>,
```

Comment:
```suggestion
      private readonly _tableElement: ElementRef<CdkTable<T>>,
```
nit: prefer `_tableElement` written out

Reply:
Done